packages/connect-react/src/index.ts

```diff
@@ -184,52 +211,108 @@ export function usePermissions(): [Permission[], LoadingStatus] {
   return useConnectSubscription<Permission[]>(callback, [])
 }
 
-export function createAppHook(appConnect: Function) {
+export function createAppHook(
+  appConnect: Function,
+  connector?: string | [string, { [key: string]: any } | undefined]
+) {
   return function useAppData<T = any>(
     app: App | null,
-    callback?: (app: App | any) => T | Promise<T>
+    callback?: (app: App | any) => T | Promise<T>,
+    dependencies?: any[]
   ): [T | null, LoadingStatus] {
-    const [result, setResult] = useState<T | null>(null)
-    const [error, setError] = useState<Error | null>(null)
-    const [loading, setLoading] = useState<boolean>(false)
+    const [{ result, error, loading }, setStatus] = useState<{
+      error: Error | null
+      loading: boolean
+      result: T | null
+    }>({
+      error: null,
+      loading: false,
+      result: null,
+    })
+
     const callbackRef = useRef<Function>((app: App) => app)
 
     useEffect(() => {
       callbackRef.current = callback || ((app: App) => app)
     }, [callback])
 
+    const appJsonRef = useRef<string>('')
+    const [updatedApp, setUpdatedApp] = useState<App | null>(null)
+    useEffect(() => {
+      const appJson = JSON.stringify(app)
+      if (appJson !== appJsonRef.current) {
+        setUpdatedApp(app)
+        appJsonRef.current = appJson
+      }
+    }, [app])
+
     useEffect(() => {
       let cancelled = false
+      let subscriptionHandler: SubscriptionHandler | null = null
 
-      setResult(null)
-      setError(null)
+      setStatus((status) => ({
+        ...status,
+        error: null,
+        result: null,
+      }))
 
       const update = async () => {
-        setLoading(true)
+        setStatus((status) => ({
+          ...status,
+          loading: true,
+        }))
 
         try {
-          const connectedApp = await appConnect(app)
+          const connectedApp = await appConnect(updatedApp, connector)
           const result = await callbackRef.current(connectedApp)
-          if (!cancelled) {
-            setLoading(false)
-            setResult(result)
+          if (cancelled) {
+            return
+          }
+
+          // Subscription
+          if (typeof result === 'function') {
+            console.log('Subscription mode')
+            subscriptionHandler = result(
+              (error: Error | null, result?: any) => {
+                if (!cancelled) {
+                  setStatus({
+                    error: error || null,
+                    loading: false,
+                    result: error ? null : result,
+                  })
+                }
+              }
+            )
+
+            // Just async data
+          } else {
+            console.log('Subscription mode')
```

Comment:
Is this log accurate?

Reply:
It’s not, and it shouldn’t have been committed either :smile: